Assets/Scripts/Data/DataModels/Missions/Mission.cs

```diff
@@ -44,6 +53,29 @@ public async Task LoadDataAsync()
         saveData = await DataUtils.LoadFileAsync<MissionSaveData>(name, FOLDER_NAME);
     }
 
+    public void UnlockIfConditionMet()
```

Comment:
Reasons why I bundled these into the same method: 

1. The OnFinancialTransaction event is parameterless, and the OnExecute event should be too (should only trigger by way of conversation node condition), so we can't overload. I could make the transaction event accept a long but then we have the same issue as before (we want to reach in and access singleton state directly).
2. I only want one method to appear in the inspector for the conversation iOnExecute events n the dialogue database. I tried [HideInInspector] but doesn't work for events (would need a custom inspector that wouldn't really be worth making). Reason for one method is so content editors can't choose the wrong one, and end up not unlocking the mission.

Was thinking a generic delegate but you can't have a void method.